rust-blas/src/math/bandmat.rs

```diff
@@ -72,17 +75,54 @@ impl<T> BandMat<T> {
         self.data.push(val);
     }
 
-    pub unsafe fn from_matrix(
-        mut mat: Mat<T>,
+    /// Converts a standard matrix into a band matrix
+    ///
+    /// TODO: write a conversion utility
+    /// DOES NOT PERFORM CONVERSION INTO BAND MATRIX FORMAT
+    ///
+    /// The actual conversion can be done by hand once one
+    /// understands the pattern of the band matrix. Say you
+    /// have an MxN band matrix in standard form. Say you 
+    /// have ku upper diagonals and kl sub diagonals. Then
+    /// the band matrix representation will have dimensions 
+    /// Nx(ku+kl+1), and will have the all of the relevant 
+    /// values of each row, concatenated starting at position
+    /// kl. An example is probably useful:
+    ///
+    /// Say our initial matrix is the following:
+    ///
+    /// [ 1 2 0 0 ]
+    /// [ 3 1 2 0 ]
+    /// [ 0 3 1 2 ]
+    /// [ 0 0 3 1 ]
+    ///
+    /// Then the band matrix representation would be:
+    ///
+    /// [ x 1 2 ]
+    /// [ 3 1 2 ]
+    /// [ 3 1 2 ]
+    /// [ 3 1 x ]
+    ///
+    /// "x" represent values that aren't relevant to the 
+    /// calculations: the values in those slots won't be
+    /// read or modified by the underlying BLAS program.
+    ///
+    pub fn from_matrix(
```

Comment:
the disclaimer and name of this func are kind of contradictory, isn't it? `band_mat.from_matrix()` should do the conversion, but I think it is a very expensive operation?

Reply:
It's a O(n*(ku+kl+1)) operation if we want constant memory overhead. And I would implement it, however it would result in dimension erasure of the original matrix. 

Perhaps we can use a matrix pointer instead of consuming the matrix? That way we don't have to worry about the memory being double-freed because of lifetime constraints.

Reply:
I know too little about the topic to say either way - I am also not sure what implications dimensionality erasure would cost us here, I am just not really keen introducing pointers anywhere :]
There is also `Cow<>` types, which might be worth investigating given you use subslices a lot?